src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
@huonw Thanks for the pointers, apparantly I was wrong.

So, now for the wording, what about putting "behaviour for iterators with more than `usize::MAX` elements is unspecified" into `# Undefined overflow`? What do you suggest?